activesupport/CHANGELOG.md

```diff
@@ -1,3 +1,34 @@
+*   Override `inspect` methods in `CombinedConfiguration`, `EnvConfiguration`, and
+    `EncryptedConfiguration` to show key names while hiding sensitive values. This prevents
+    accidental exposure of secrets in console or logs while still providing useful debugging
+    information about what configuration keys are available.
+
+    Before:
+
+    ```ruby
+    Rails.app.creds
+    #<ActiveSupport::CombinedConfiguration:0x0000000122b45870
+     @configurations=[
+       #<ActiveSupport::EnvConfiguration:0x0000000122b465b8
+        @envs={"SOME_SECRET" =>  "secret_value", ...}>,
+       #<ActiveSupport::EncryptedConfiguration:0x00000000007af0>]>
+    ```
+
+    After:
+
+    ```ruby
+    Rails.app.creds
+    #<ActiveSupport::CombinedConfiguration:0x00000000006490 keys=[:rails_env, :some_secret, ..., :secret_key_base, :stripe_webhook_secret]>
+
+    Rails.app.credentials
+    #<ActiveSupport::EncryptedConfiguration:0x00000000008cd0 keys=[:secret_key_base, :stripe_webhook_secret]>
+
+    Rails.app.envs
+    #<ActiveSupport::EnvConfiguration:0x0000000000a6a0 keys=[:rails_env, :some_secret, :path, :home, ...]>
+    ```
+
+    *Emmanuel Hayford*
```

Comment:
Since this is an unreleased feature, we don't need a CHANGELOG entry. But we can amend the other entries to note that these new configuration objects are inspect safe.